src/tqec/computation/correlation.py

```diff
@@ -169,3 +170,31 @@ def external_stabilizer(self, io_ports: list[int]) -> str:
             for port in io_ports
         ]
         return "".join(paulis)
+
+    def external_stabilizer_on_graph(self, graph: BlockGraph) -> str:
+        """Get the Pauli operator supported on the input/output ports of the graph.
+        The ports are ordered according to their labels.
+
+        Args:
+            g: The block graph to consider.
+
+        Returns:
+            The Pauli operator supported on the input/output ports of the graph.
+        """
+        port_labels = graph.ordered_ports
+        port_positions = [graph.ports[p] for p in port_labels]
+        zx = graph.to_zx_graph()
+        zx_ports = [zx.p2v[p] for p in port_positions]
```

Comment:
Note that here `p2v` is re-computed at each call. Maybe cache it in a variable to avoid that? 
```suggestion
        p2v = zx.p2v
        zx_ports = [p2v[p] for p in port_positions]
```